test/GetVlanInterfaces_spec.rb

```diff
@@ -8,21 +8,57 @@
   end
   
   describe "#GetVlanInterfaces" do
-    context "with valid arguments" do
-      it "returns map containing info about vlan interfaces per netcard" do
+    context "with a list of netcards and a list of corresponding VLANs as arguments" do
+      it "returns info about VLAN interfaces per netcard (without identical VLAN IDs)" do
         expect(@fcoe.GetVlanInterfaces(["eth0", "eth1", "eth2"],
                                        ["eth0     | 200  | 00:0d:ec:a2:ef:00",
                                         "eth0     | 300  | 00:0d:ec:a2:ef:01",
-                                        "eth2     | 200  | 00:0d:ec:a2:ef:02" ])).to eq(
+                                        "eth0     | 300  | 00:0d:ec:a2:ef:02",
+                                        "eth1     | 400  | 00:ef:af:34:12:ae",
+                                        "eth1     | 400  | 00:ef:af:34:12:af",
+                                        "eth2     | 200  | 00:0d:ec:a2:ef:03" ])).to eq(
                       {"eth0" => [{ "vlan" => "200", "fcf" => "00:0d:ec:a2:ef:00" },
                                   { "vlan" => "300", "fcf" => "00:0d:ec:a2:ef:01" }],
-                       "eth2" => [{ "vlan" => "200", "fcf" => "00:0d:ec:a2:ef:02" }]
+                       "eth1" => [{ "vlan" => "400", "fcf" => "00:ef:af:34:12:ae" }],
+                       "eth2" => [{ "vlan" => "200", "fcf" => "00:0d:ec:a2:ef:03" }]
                       })
       end
     end
   end
+
+  describe "#GetVlanInterfaces" do
```

Comment:
It is not other method that you are describing but other context, so it's not needed.